New HTML widget documentation (#3901)
SoftwareAG/c8y-docs: content/cockpit/widgets-collection.md

You must enable at least one data point in each section to create the "Info gauge" widget.

### HTML {#html}

> The "HTML" widget displays user-defined content that can be formatted using HTML and dynamically populated with data from the current device. Additionally, you can switch the widget into advanced mode, which allows you to build complex web components with JavaScript code.

Reviewer: "current device" -> selected asset or device

**Parameters to configure**

* **Target assets or devices**: Select the objects for which optional HTML expressions are evaluated.

> * **Asset properties**: Based on the target asset, you can copy properties in the **Asset properties** tab, which you can paste into the code shown under settings.

Reviewer: ```suggestion
* **Asset properties**: In the **Asset properties** section, you can copy the properties of the selected asset and paste them into the code editor under the **Settings**.
```

1. **Normal mode**: You can apply HTML and CSS while adding properties as template literals. You can use simple expressions such as:
   `${this.c8yContext ? this.c8yContext.name : 'No device selected'}`. The `${this.c8yContext}` variable always refers to the selected target asset.

2. **Advanced mode**: When enabled, you can build complex web components using the Lit framework. You can import supported ECMAScript modules. By default, leaflet, echarts, fetch, and lit are provided. Whatever is rendered in the web component will be displayed to the end user. Additional requests can be performed by importing the fetch library. The following shows the available imports:

   ```javascript
   import { LitElement, html, css } from 'lit';
   import { styleImports } from 'styles';
   import { L } from 'leaflet';
   import * as echarts from 'echarts';
   import { fetch } from 'fetch'; // Use this instead of default fetch to avoid potential issues
   ```

**Styling and security considerations**

When using styles, global styles can be applied if encapsulation is not enabled. Styles should always use CSS variables and tokens to ensure compatibility with dark mode and custom brandings. 

> By default, the normal HTML widget is sanitized for security, while in advanced mode the developer is responsible for proper sanitization. You can modify these settings in the application configuration.

Reviewer: "these settings" refer to css encapsulation and sanitization? What "application configuration" do you refer to?

### HTML {#html}

The "HTML" widget displays user-defined content that can be formatted using HTML and dynamically populated with data from the current device. Additionally, you can switch the widget into advanced mode, which allows you to build complex web components with JavaScript code.

**Parameters to configure**

* **Target assets or devices**: Select the objects for which optional HTML expressions are evaluated.

> * **Asset properties**: Based on the target asset, in the "Asset properties" tab you can copy properties which you can paste into the code shown under settings.

Reviewer 2: ```suggestion
* **Asset properties**: Based on the target asset, you can copy properties in the **Asset properties** tab, which you can paste into the code shown under settings.
```

The widget offers two distinct modes:

1. **Normal mode**: You can apply HTML and CSS while adding properties as template literals. You can use simple expressions such as:
   `${this.c8yContext ? this.c8yContext.name : 'No device selected'}`. The `${this.c8yContext}` variable always refers to the selected target asset.

> 2. **Advanced mode**: When enabled, you can build complex web components using the Lit framework. You can import supported ECMAScript modules. By default, we provide leaflet, echarts, fetch, and lit. Whatever is rendered in the web component will be displayed to the end user. Additional requests can be performed by importing the fetch library. The following shows the available imports:

Reviewer 2: ```suggestion
2. **Advanced mode**: When enabled, you can build complex web components using the Lit framework. You can import supported ECMAScript modules. By default, leaflet, echarts, fetch, and lit are provided. Whatever is rendered in the web component will be displayed to the end user. Additional requests can be performed by importing the fetch library. The following shows the available imports:
```

   ```javascript
   import { LitElement, html, css } from 'lit';
   import { styleImports } from 'styles';
   import { L } from 'leaflet';
   import * as echarts from 'echarts';
   import { fetch } from 'fetch'; // Use this instead of default fetch to avoid potential issues
   ```

**Styling and security considerations**

When using styles, global styles can be applied if encapsulation is not enabled. Styles should always use CSS variables and tokens to ensure compatibility with dark mode and custom brandings. 

By default, the normal HTML widget is sanitized for security, while in advanced mode the developer is responsible for proper sanitization. You can modify these settings in the application configuration.

A simple example looks like this:

![HTML widget](/images/users-guide/cockpit/cockpit-widget-html.png)

**Legacy widget compatibility**

{{< c8y-admon-important >}}

> Existing widgets based on AngularJS will automatically fall back to legacy mode, which maintains backward compatibility and allows these widgets to continue functioning. This fallback displays JavaScript code that enables legacy widget execution. However, it is strongly recommended to migrate these widgets to the new Lit-based framework as soon as possible, since AngularJS support is deprecated and will be removed in future releases.

Reviewer 2: ```suggestion
Existing widgets based on AngularJS will automatically fall back to legacy mode, which maintains backward compatibility and allows these widgets to continue functioning. This fallback displays JavaScript code that enables legacy widget execution. However, it is strongly recommended to migrate these widgets to the new Lit-based framework as soon as possible, since AngularJS support is deprecated.
```

Reviewer 2: I would be careful with "future" mentions, as they tend not to be removed when the future has arrived. I would simply drop that part of the sentence.
